bpf/lib/encap.h

```diff
@@ -295,5 +295,76 @@ encap_and_redirect_netdev(struct __ctx_buff *ctx, struct tunnel_key *k,
 }
 #endif /* TUNNEL_MODE */
 
+#if defined(ENABLE_DSR) && DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+static __always_inline int
+__encap_with_nodeid_opt(struct __ctx_buff *ctx, __u32 tunnel_endpoint,
```

Comment:
IMO this code should not be referring to `nodeid` or `node_id` now that we have a concept called node id. My recommendation for this PR would be replacing existing references to `nodeid`/`node_id` with something like `tunnel_peer`, and to rename these functions as `*encap_with_dsr_opt(...)`:

```suggestion
__encap_with_dsr_opt(struct __ctx_buff *ctx, __u32 tunnel_endpoint,
```

cc @julianwiedmann @pchaigno it looks like we're starting to mix up different concepts in code like this, partly related to the way that `ctx_set_encap_info()` names the *tunnel peer* parameter (which is a destination IPv4 address) as `node_id`. I think that parameter should just be renamed in existing code.

I realize that this PR is just following the existing patterns and the nodeid naming in this code predates Paul's recent work, but that's all the more reason to be vigilant about naming in order to avoid confusing two different concepts by using the same name for both.

Reply:
cf. Commit 814d3c797589 ("ipcache: Populate ipcache with node IDs")

Reply:
(I would be OK with accepting this code as-is then just following up separately to cover the variable/function rename in various places across the codebase)

Reply:
Yes, the naming of the encap helpers hasn't aged well. And there's more churn coming (eg. adjustments needed for in-XDP encap). So let's tackle the rename separately.